src/codegen_call_native.cpp

```
@@ -199,3 +209,217 @@ llvm::Value *CodeGen::emitTableDrivenNativeCall(
+    if (libIt == native_lib_index_.end())
+        return nullptr;
+
+    // Emit args once, then try each candidate library for a type match.
+    // This avoids committing to the first arity match when a different
+    // library's signature may match the actual argument types.
+    std::vector<llvm::Value *> args;
+    for (size_t i = 0; i < e.args.size(); i++)
+        args.push_back(emitExpr(*e.args[i]));
+
+    const NativeFnSignature *matchedSig = nullptr;
+    std::string matchedPackage;
```

Clang (14.0.6)

[warning] variable 'matchedPackage' is not initialized

(cppcoreguidelines-init-variables)